### PR TITLE
PRO-1341 preserve the hash when updating query strings

### PR DESCRIPTION
**Pull request by boutell**
Also:

* eslint fixes.
* Don't manually chop off old query strings before calling, `addQueryToUrl` is smart enough to do that.

### CODE REVIEW

**Comment by boutell** [comment]
More importantly I guess, the existing logic failed to add the query string properly because it added it after the # and it never reached the server.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,8 @@
 ### Fixes
 
 - The `name` option to widget modules, which never worked in 3.x, has been officially removed. The name of the widget type is always the name of the module, with the `-widget` suffix removed.
+- In-context editing works properly when the current browser URL has a `#` part, enabling the use of the `#` part for project-specific work. Thanks to [https://stepanjakl.com/](Štěpán Jákl) for reporting the issue.
```
Comment (abea) [inline CHANGELOG.md:12]: I think we can call it a hash.

**Review by abea — CHANGES_REQUESTED** [review]
One changelog note. The term "hash" is part of the location API.

**Review by abea — CHANGES_REQUESTED** [review]
Still the changelog bit.

**Review by abea — APPROVED** [review]

**Pull request merged**